flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/RexNodeJsonSerdeTest.java

```diff
@@ -0,0 +1,831 @@
+        return serdeContext;
+    }
+
+    private static void dropPermanentFunction(SerdeContext serdeContext) {
+        serdeContext
+                .getFlinkContext()
+                .getFunctionCatalog()
+                .dropCatalogFunction(UNRESOLVED_FUNCTION_CAT_ID, false);
+    }
+
+    private static void modifyPermanentFunction(SerdeContext serdeContext) {
+        dropPermanentFunction(serdeContext);
+        serdeContext
+                .getFlinkContext()
+                .getFunctionCatalog()
+                .registerCatalogFunction(UNRESOLVED_FUNCTION_CAT_ID, SER_UDF_CLASS_OTHER, false);
+    }
+
+    private static void registerTemporaryFunction(SerdeContext serdeContext) {
+        serdeContext
+                .getFlinkContext()
+                .getFunctionCatalog()
+                .registerTemporaryCatalogFunction(
+                        UNRESOLVED_FUNCTION_CAT_ID, NON_SER_FUNCTION_DEF_IMPL, false);
+    }
+
+    private JsonNode serialize(
+            SerdeContext serdeContext, ContextResolvedFunction contextResolvedFunction)
+            throws Exception {
+        final byte[] actualSerialized =
+                createObjectWriter(serdeContext)
+                        .writeValueAsBytes(
+                                createFunctionCall(serdeContext, contextResolvedFunction));
+        return getObjectMapper().readTree(actualSerialized);
+    }
+
+    private ContextResolvedFunction deserialize(SerdeContext serdeContext, JsonNode node)
+            throws IOException {
+        final RexNode actualDeserialized =
+                createObjectReader(serdeContext).readValue(node, RexNode.class);
+        return ((BridgingSqlFunction) ((RexCall) actualDeserialized).getOperator())
+                .getResolvedFunction();
+    }
+
+    private static SerdeContext contradictingSerdeContext() {
+        // these contradicting options should not have an impact on temporary or anonymous functions
+        return serdeContext(CatalogPlanCompilation.IDENTIFIER, CatalogPlanRestore.ALL_ENFORCED);
+    }
+
+    private static class FunctionProvidingModule implements Module {
+        private static final FunctionProvidingModule INSTANCE = new FunctionProvidingModule();
+
+        @Override
+        public Set<String> listFunctions() {
+            return Collections.singleton(FUNCTION_NAME);
+        }
+
+        @Override
+        public Optional<FunctionDefinition> getFunctionDefinition(String name) {
+            if (name.equalsIgnoreCase(FUNCTION_NAME)) {
+                return Optional.of(NON_SER_FUNCTION_DEF_IMPL);
+            }
+            return Optional.empty();
+        }
+    }
+
+    // --------------------------------------------------------------------------------------------
+    // Test functions
+    // --------------------------------------------------------------------------------------------
+
+    /** Serializable function. */
+    @SuppressWarnings("unused")
```

Comment:
Here and below, why do we need to suppress `unused`?

Reply:
Ah, it's the variables for the methods throwing exceptions, then maybe nit: to move the `@SuppressWarnings` over those methods.